test/ex_riak/pb_socket_test.exs

```diff
@@ -46,6 +46,32 @@ defmodule ExRiak.PBSocketTest do
     end
   end
 
+  describe "get_binary_index_eq/5" do
+    test "when the index has values", %{conn: conn} do
+      value = "world"
+
+      index = "things i've said hello to"
+      index_value = "planets"
+
+      obj = Object.new(basic_bucket(), :undefined, value, 'text/plain')
+
+      md =
+        obj
+        |> ExRiak.Object.get_update_metadata!
+        |> ExRiak.Object.set_secondary_index([{{:binary_index, index}, [index_value]}])
+
+      obj = ExRiak.Object.update_metadata(obj, md)
+
+      {:ok, key} = PBSocket.put(conn, obj)
+      {:ok, another_key} = PBSocket.put(conn, obj)
```

Comment:
This is very cool. I didn't realize  `PBSocket.put/2` returned a generated key. I just fixed the typespecs and updated the docs in 3d39490.